drivers/char/hw_random/bcm2835_rng_rust.rs

```diff
@@ -20,15 +26,41 @@ module! {
     license: b"GPL v2",
 }
 
+struct RngDevice;
+
+impl FileOpener<()> for RngDevice {
+    fn open(_state: &()) -> Result<Self::Wrapper> {
+        Ok(Box::try_new(RngDevice)?)
+    }
+}
+
+impl FileOperations for RngDevice {
+    kernel::declare_file_operations!(read);
+
+    fn read<T: IoBufferWriter>(&self, _: &File, data: &mut T, offset: u64) -> Result<usize> {
+        // Succeed if the caller doesn't provide a buffer or if not at the start.
+        if data.is_empty() || offset != 0 {
+            return Ok(0);
+        }
+
+        data.write(&0_u32)?;
```

Comment:
Maybe add a FIXME just in case?

Reply:
Thanks for the review! While the PR x/6 effort is ongoing, the driver is a continual WIP. I'm just adding functionality, as it becomes available in the Rust kernel core.

Do you think there is merit in placing a "warning this is a work in progress, not fully functional" comment on top of the driver?

Reply:
Maybe?

Reply:
Sounds reasonable to me. I'll add it.